### PR TITLE
Add earlgrey_gem.yml & Fix rubocop

### LINKED ISSUE #668: Allow disabling EarlGrey.swift installation via Cocoapods
```ruby
target 'SoGrey' do
    project 'Frameworks/SoGrey/SoGrey.xcodeproj'
    pod 'EarlGrey', '~> 1.1'
end

abstract_target 'common_ui_tests' do
    target 'TeacherUITests' do
      pod 'SwiftProtobuf', '~> 0.9.904'
      pod 'EarlGrey', '~> 1.1'
      project 'rn/Teacher/ios/Teacher.xcodeproj'
    end
end
```

- https://github.com/instructure/instructure-ios/blob/master/Podfile

`EarlGrey.swift` is provided by a shared framework `SoGrey` so we don't want `EarlGrey.swift` from the gem. You can see in the podfile, `EarlGrey.swift` is added to the `SoGrey` target. Then we have a `TeacherUITests` that should not have `EarlGrey.swift`

- https://github.com/instructure/instructure-ios/blob/master/Frameworks/SoGrey/SoGrey/EarlGrey.swift

There should be a way to tell EarlGrey to not add EarlGrey.swift on certain targets.

### PR DESCRIPTION
**Pull request by bootstraponline**
Update Ruby version requirement to 2.1 (Rubocop dropped 2.0). Both 2.0 and 2.1 are end of life Ruby versions.

- https://www.ruby-lang.org/en/downloads/

Fix #668

### CODE REVIEW

**Comment by bootstraponline** [comment]
image
Ruby job passed on Travis.

**Comment by khandpur** [comment]
@wuhao5 could you please review this?

File: gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+TeacherUITests:
```
Comment (wuhao5) [inline gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml]: which yml will be used for the end user? can you give an example?

File: .rubocop.yml [hunk 2]
```diff
@@ -12,8 +12,10 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 #
+AllCops:
+  TargetRubyVersion: 2.1
 Metrics/LineLength:
-  Max: 100
+  Max: 130
```
Comment (wuhao5) [inline .rubocop.yml]: is this only for not breaking the one liner?

File: gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml (hunk 1 above)
Comment (tirodkar) [inline gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml]: Change to something general like - SampleEarlGreySwiftTests ?

**Comment by bootstraponline** [comment]
Thanks for the comments. I'll update the PR tomorrow.

File: .rubocop.yml (hunk 2 above)
Comment (bootstraponline) [reply .rubocop.yml]: 100 is too small.

```
Offenses:

Demo/EarlGreyExample/Podfile:23:8: C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
target "EarlGreyExampleTests" do
       ^^^^^^^^^^^^^^^^^^^^^^
Demo/EarlGreyExample/Podfile:30:8: C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
target "EarlGreyExampleSwiftTests" do
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Demo/EarlGreyExample/Podfile:36:1: C: Layout/TrailingBlankLines: 1 trailing blank lines detected.
gem/spec/earlgrey_yaml_spec.rb:6:101: C: Metrics/LineLength: Line is too long. [105/100]
    allow(podfile).to receive(:defined_in_file) { File.join(__dir__, '/fixtures/earlgrey_yaml/Podfile') }
                                                                                                    ^^^^^
gem/lib/earlgrey/cli.rb:24:101: C: Metrics/LineLength: Line is too long. [110/100]
    method_option :scheme,  aliases: '-s', type: :string, required: false, desc: 'EarlGreyTestTarget.xcscheme'
                                                                                                    ^^^^^^^^^^
gem/lib/earlgrey/extensions/analyzer_extensions.rb:35:101: C: Metrics/LineLength: Line is too long. [119/100]
          EarlGrey.add_earlgrey_copy_files_script native_target, framework_ref if config[EarlGreyYaml::ADD_BUILD_PHASE]
                                                                                                    ^^^^^^^^^^^^^^^^^^^
gem/lib/earlgrey/extensions/analyzer_extensions.rb:36:101: C: Metrics/LineLength: Line is too long. [109/100]
          EarlGrey.add_earlgrey_framework native_target, framework_ref if config[EarlGreyYaml::ADD_FRAMEWORK]
                                                                                                    ^^^^^^^^^
gem/lib/earlgrey/extensions/earlgrey_yaml.rb:19:101: C: Metrics/LineLength: Line is too long. [116/100]
    abort "ERROR: earlgrey_gem.yml references missing targets: #{missing_targets}".red unless missing_targets.empty?
                                                                                                    ^^^^^^^^^^^^^^^^
gem/lib/earlgrey/extensions/earlgrey_yaml.rb:30:101: C: Metrics/LineLength: Line is too long. [105/100]
    abort "ERROR: earlgrey_gem.yml contains unknown keys: #{unknown_keys}".red unless unknown_keys.empty?
                                                                                                    ^^^^^
gem/lib/earlgrey/configure_earlgrey.rb:319:101: C: Metrics/LineLength: Line is too long. [129/100]
        settings[HEADER_SEARCH_PATHS] << CARTHAGE_HEADERS_IOS unless settings[HEADER_SEARCH_PATHS].include?(CARTHAGE_HEADERS_IOS)
                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gem/lib/earlgrey/configure_earlgrey.rb:345:101: C: Metrics/LineLength: Line is too long. [127/100]
      target.frameworks_build_phase.add_file_reference framework_ref, true unless linked_frameworks.include? EARLGREY_FRAMEWORK
                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
gem/lib/earlgrey/configure_earlgrey.rb:384:101: C: Metrics/LineLength: Line is too long. [107/100]
        raise "Unable to locate #{swift_fallback} file at path #{src_swift}." unless File.exist?(src_swift)
                                                                                                    ^^^^^^^
```

File: gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml (hunk 1 above)
Comment (bootstraponline) [reply gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml]: Done

Comment (bootstraponline) [reply gem/spec/fixtures/earlgrey_yaml/earlgrey_gem.yml]: I added earlgrey_gem.yml to EarlGreyExample.

**Comment by bootstraponline** [comment]
I've addressed all the comments. Let me know if there's any more feedback.

File: Demo/EarlGreyExample/earlgrey_gem.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+EarlGreyExampleSwiftTests:
```
Comment (wuhao5) [inline Demo/EarlGreyExample/earlgrey_gem.yml]: this is cool, didn't know about this setting, could you also update the doc?

Comment (bootstraponline) [reply Demo/EarlGreyExample/earlgrey_gem.yml]: updated the readme

**Comment by wuhao5** [comment]
Thanks! @bootstraponline LGTM.

**Comment by wuhao5** [comment]
@bootstraponline there is a travis failure, do you mind taking a look?

**Comment by bootstraponline** [comment]
> @bootstraponline there is a travis failure, do you mind taking a look

I think Travis itself is the failure. The CI setup we have for EarlGrey has been failing for years. I'm spending the majority of my time waiting for builds that don't run locally.

For an automation project, the CI shouldn't be such a mess with builds that can't run locally and a solid 0% pass rate.

The CI system needs to be replaced. The process for getting feedback and resolving issues is too slow.

**Comment by bootstraponline** [comment]
It looks like `EarlGrey.swift` isn't getting created.

**Comment by bootstraponline** [comment]
Issue should be fixed now.

**Comment by bootstraponline** [comment]
> Error Domain=IDETestOperationsObserverErrorDomain Code=14 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at 
>
> https://api.travis-ci.org/v3/job/339106186/log.txt

this seems like general flakiness?

**Comment by wuhao5** [comment]
thanks @bootstraponline , I think you will need to sync up with the latest HEAD as there is some merges lately.

**Comment by bootstraponline** [comment]
`Compiling[0m EarlGrey.swift` is failing. Pretty sure that has nothing to do with the Ruby 💎 .

**Comment by tirodkar** [comment]
@bootstraponline know any reason why it might be failing? None of us seem to have seen it before either.

**Comment by bootstraponline** [comment]
> @bootstraponline know any reason why it might be failing? None of us seem to have seen it before either.

No idea... Travis seems like a lost cause. Appium recently moved to Circle CI. EarlGrey needs to move to something where we can easily reproduce failures locally to debug.

**Comment by bootstraponline** [comment]
Travis is failing right now on 100% of the EarlGrey open PRs.  😭

**Pull request merged**